src/end-game.js

```diff
@@ -0,0 +1,98 @@
+//This file is created to determine end game/sanbox for writing new code
+
+const VALUES ={
+    x: "x",
+    o: "o",
+    incomplete: ""
+    noWinner: "no winner"
+}
+
+//get rows from board, actually returns columns
+const getRowsFromBoard = (board) =>{
+    const rowlenght = Math.sqrt(board.length);
+    const rows = [];
+    for (let x=0;x<rowlenght;x++){
+        rows.push([]);
+    }
+    for (let x=0;x<board.length;x++){
+        const currentValue = board[x];
+        const currentRowIndex = Math.floor(x/rowlenght);
+        const currentRow = rows[currentRowIndex];
+        currentRow.push(currentValue)
+
+    }
+    return rows
+}
+
+//get columns
+const getColumnsFromBoard = (board) =>{
+    const rowlenght = Math.sqrt(board.length);
+    const columns = [];
+    for (let x=0;x<rowlenght;x++){
+        colums.push([]);
+    }
+    for (let x=0;x<board.length;x++){
+        const currentValue = board[x];
+        const currentColumnIndex = x%rowlenght;
+        const currentColumn = rows[currentColumnIndex];
+        currentColumn.push(currentValue)
+
+    }
+    return columns
+
+
+};
+//get diagonal from top left to bottom right
+const getDiagonal1FromBoard = (board) => {
+    const rowlength = Math.sqrt(board.length);
+    const diagonal1 = [];
+    for (let x = 0; x < rowlength; x++) {
+        diagonal1.push([]);
+    }
+
+    for (let x = 0; x < board.length; x++) {
```

Comment:
I think it might be better to calculate an array of indexes for the diagonal instead of going over the whole board.

In the left to right case for 3 by 3, you need the array `[0, 4, 8]`. 4x4 would be `[0,4,8,15]` etc...